lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -498,7 +535,11 @@ def get_students_problem_grades(request, csv=False):
         module_state_key=module_state_key,
         module_type__in=PROB_TYPE_LIST,
         grade__isnull=False,
-    ).values('student__username', 'student__profile__name', 'grade', 'max_grade').order_by('student__profile__name')
+    ).values('course_id','student__id','student__username', 'student__profile__name', 'grade', 'max_grade').order_by('student__profile__name')
+
+    # here we are using the first record to get course_id.
```

Comment:
same comment here about trying to get this from the request, and therefore the view that renders the page that issues the request.